src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,411 @@
+
+            // TODO: I think we can skip this commit and decrease the tree operations this way
+            _stateProvider.Commit(_specProvider.GetSpec(stateUpdate.Number), txTracer.IsTracingState ? txTracer : null);
+
+            long unspentGas = gasLimit - intrinsicGas;
+            long spentGas = gasLimit;
+
+            int stateSnapshot = _stateProvider.TakeSnapshot();
+            int storageSnapshot = _storageProvider.TakeSnapshot();
+
+            _stateProvider.SubtractFromBalance(sender, value, spec);
+            byte statusCode = StatusCode.Failure;
+            TransactionSubstate substate = null;
+
+            try
+            {
+                if (entry.IsValidDeploymentEntry)
+                {
+                    recipient = Address.OfContract(sender, _stateProvider.GetNonce(sender) - 1);
+                    if (_stateProvider.AccountExists(recipient))
+                    {
+                        if ((_virtualMachine.GetCachedCodeInfo(recipient)?.MachineCode?.Length ?? 0) != 0 || _stateProvider.GetNonce(recipient) != 0)
+                        {
+                            if (_logger.IsEnabled(LogEventLevel.Verbose))
+                            {
+                                _logger.Verbose($"Contract collision at {recipient}"); // the account already owns the contract with the code
+                            }
+
+                            throw new TransactionCollisionException();
+                        }
+
+                        _stateProvider.UpdateStorageRoot(recipient, Keccak.EmptyTreeHash);
+                    }
+                }
+
+                bool isPrecompile = recipient.IsPrecompiled(spec);
+
+                ExecutionEnvironment env = new ExecutionEnvironment();
+                env.Value = value;
+                env.TransferValue = value;
+                env.Sender = sender;
+                env.CodeSource = recipient;
+                env.ExecutingAccount = recipient;
+                env.CurrentBlock = stateUpdate;
+                env.GasPrice = gasPrice;
+                env.InputData = data ?? new byte[0];
+                env.CodeInfo = isPrecompile ? new CodeInfo(recipient) : machineCode == null ? _virtualMachine.GetCachedCodeInfo(recipient) : new CodeInfo(machineCode);
+                env.Originator = sender;
+
+                ExecutionType executionType = entry.IsValidDeploymentEntry ? ExecutionType.Create : ExecutionType.Call;
+                using (VmState state = new VmState(unspentGas, env, executionType, isPrecompile, true, false))
+                {
+                    substate = _virtualMachine.Run(state, txTracer);
+                    unspentGas = state.GasAvailable;
+                }
+
+                if (substate.ShouldRevert || substate.IsError)
+                {
+                    if (_logger.IsEnabled(LogEventLevel.Verbose)) _logger.Verbose("Restoring state from before transaction");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)